instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/snippet/ServletOutputStreamInjectionHelper.java

```diff
@@ -42,8 +42,16 @@ public static boolean handleWrite(
     state.setAlreadyInjected(); // set before write to avoid recursive loop
     out.write(original, off, i + 1);
     try {
-      byte[] snippetBytes = SnippetHolder.getSnippetBytes(state.getCharacterEncoding());
+      byte[] snippetBytes = ExperimentalSnippetHolder.getSnippetBytes(state.getCharacterEncoding());
+      if (state.getWrapper().isCommitted()) {
+        // header already set and sent, don't inject
```

Comment:
maybe we can do slightly better, and still inject even if it was committed, but only when content length was not set